regression/contracts/assigns_enforce_23/main.c

```diff
@@ -0,0 +1,41 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef struct
+{
+  uint8_t *buf;
+  size_t size;
+} blob;
+
+void foo(blob *b, uint8_t *value)
+  // clang-format off
+__CPROVER_requires(b->size == 5)
+__CPROVER_assigns(__CPROVER_POINTER_OBJECT(b->buf))
+__CPROVER_assigns(__CPROVER_POINTER_OBJECT(value))
```

Comment:
This overloaded use of `__CPROVER_POINTER_OBJECT` will confuse. I would advise to use a different identifier.